### ISSUE TITLE
Install package manually

### ISSUE THREAD

**Issue by vsheg** [rstudio/tinytex#377]
Hello, I've just spent a lot of time on how to install my custom package globally for `tinytex` (CRAN's v0.40), hope this short instruction will be indexed by Google, and whoever needs it will find it. 

I'm using mac.

1. Create `texmf` folder
```bash
mkdir ~/Library/texmf
```
2. Add [necessary](https://tex.stackexchange.com/questions/8357/how-to-have-local-package-override-default-package) directories and copy files into there, e.g.:
```
mkdir -p ~/Library/texmf/tex/latex/
cp my_latex_package.sty ~/Library/texmf/tex/latex/
```
3. Add to TeX's "search PATH"
```bash
tlmgr conf auxtrees add ~/Library/texmf
```

Now, this package will be available for `\usepackage{my_latex_package}` globally

P.S. I guess `tinytex` docs should contains this, or this should be done during the `tinytex::install_tinytex()`

**Comment by yihui** [comment]
I'm not sure what we should do here. I guess this won't be an FAQ since I don't think it would be frequently asked. In most cases, I think users will just install packages from CTAN. If a custom `.sty` is needed, users can either put it under a texmf tree, or simply put it under the same directory as the `.tex` file (no need to "install" it). The former is useful only when this `.sty` is needed for multiple `.tex` documents under different directories in the system.

One caveat is that when you upgrade or reinstall TinyTeX, you will have to repeat the 3rd step above.

I think this Github issue should be good enough for now for people who want to know how to add a custom package to TeX Live. I'm in the same hope as @vsheg that this will be indexed by search engines, and users will be able to find it in future. Thanks for sharing!

BTW, @cderv `tinytex::r_texmf` is an example of adding a texmf tree so that TeX Live will be aware of it.

**Comment by vsheg** [comment]
The default `tinytex`'s auxtree (see: `~/Library/TinyTeX/texmf.cnf`)
```
TEXMFAUXTREES = /opt/homebrew/Cellar/r/4.2.0/lib/R/share/texmf
```
is unobvious, it depends on R's version and a package manager. Installing custom or patched versions of packages in `tinytex`'s guts
```bash
$ mktexlsr
mktexlsr: Updating ~/Library/TinyTeX/texmf-config/ls-R...
mktexlsr: Updating ~/Library/TinyTeX/texmf-dist/ls-R...
mktexlsr: Updating ~/Library/TinyTeX/texmf-local/ls-R...
mktexlsr: Updating ~/Library/TinyTeX/texmf-var/ls-R...
```
is not something called "a good practice" and changes do not preserve between LaTeX distributions and `tinytex` installations.
***
@yihui I still think that `tinytex` should search packages in `~/Library/texmf` by default, because it
1. is expected by users (especially for those who came from TeX Live);
2. does not require any complex or tricky support;
3. allows using "package installation" manuals from the internet;
4. is transferable between different LaTeX distributions, it should just work.

And manual editing of the `auxtree` is not something everyone should know.
***
The case of using own custom packages is quite common, I think. Many create a standard preamble for all their documents (my case) and connect it as a package `\usepackage{my_preamble}`, some packages (e.g. `mol2chemfig`) need Python or Lua modules to work, so they must be installed manually or by using some installation script that may know nothing about `tinytex`'s `texmf` tree.

**Comment by yihui** [comment]
> The default `tinytex`'s auxtree (see: `~/Library/TinyTeX/texmf.cnf`)
> 
> ```
> TEXMFAUXTREES = /opt/homebrew/Cellar/r/4.2.0/lib/R/share/texmf
> ```
> 
> is unobvious, it depends on R's version and a package manager. Installing custom or patched versions of packages in `tinytex`'s guts
> 
> ```shell
> $ mktexlsr
> mktexlsr: Updating ~/Library/TinyTeX/texmf-config/ls-R...
> mktexlsr: Updating ~/Library/TinyTeX/texmf-dist/ls-R...
> mktexlsr: Updating ~/Library/TinyTeX/texmf-local/ls-R...
> mktexlsr: Updating ~/Library/TinyTeX/texmf-var/ls-R...
> ```
> 
> is not something called "a good practice" and changes do not preserve between LaTeX distributions and `tinytex` installations.

I agree with the disadvantages you mentioned. However, my intention is not that R's texmf directory should be the *default* auxtree, but is one of the auxtrees that TeX Live should be aware of (for R users). Users can add as many auxtrees as they want.

> @yihui I still think that `tinytex` should search packages in `~/Library/texmf` by default

I agree. Perhaps I should not define `TEXMFHOME` here: https://github.com/rstudio/tinytex/blob/5231b8ecb4d7f7eeb63ed17a5750f83f03c637ab/tools/install-base.sh#L35
https://github.com/rstudio/tinytex/blob/5231b8ecb4d7f7eeb63ed17a5750f83f03c637ab/tools/install-windows.bat#L27

and just let TeX Live search its default `TEXMFHOME` directory, which I believe is something like `~/Library/texmf` on macOS, and `~/texmf` on Linux, etc., but I can't find the canonical documentation for the default value of `TEXMFHOME` now.

**Comment by cderv** [comment]
> but I can't find the canonical documentation for the default value of TEXMFHOME now.

This ? https://tug.org/texlive/doc/texlive-en/texlive-en.html#x1-250003.2.3

> TEXMFHOME is the recommended location for personal macro ﬁles or packages. **The default value is `~/texmf` (`~/Library/texmf` on Macs)**. In contrast to TEXDIR, here a `~` is preserved in the newly-written conﬁguration ﬁles, since it usefully refers to the home directory of the user running TeX. It expands to $HOME on Unix and %USERPROFILE% on Windows. Special redundant note: TEXMFHOME, like all trees, must be organized according to the TDS, or ﬁles may not be found.

**Comment by yihui** [comment]
Thanks! I just removed TinyTeX's `TEXMFHOME` configuration, so that it will just use TeX Live's default setting.
